src/main/host/process.c

```diff
@@ -581,6 +584,12 @@ static void _process_start(Process* proc) {
     worker_setActiveProcess(NULL);
     worker_setActiveThread(NULL);
 
+    if (proc->pause_for_debugging) {
```

Comment:
Optional - moving this between `vfork` and `exec` in `thread_preload.c` would let the user attach before early initialization. We'd need to be careful here to use only "async signal safe" functionality; probably `write` and `read` rather than the higher level libc functions.

If so, probably unnecessary to also implement in thread_ptrace.c, since we won't support ptrace-attaching externally anyway. (Maybe even better to explicitly error if this feature is enabled with --interpose-method=ptrace, but maybe not even worth bothering with since it'll be removed soonish...)

Reply:
I'll leave something like this for a future PR. I'm still hesitant about adding additional code after the `vfork()`.